src/misc/setup.cpp

```diff
@@ -63,32 +63,38 @@ Value &Value::copy(const Value &in)
 	return *this;
 }
 
-void Value::plaincopy(const Value &in)
+void Value::plaincopy(const Value& in)
 {
 	type    = in.type;
 	_int    = in._int;
 	_double = in._double;
-	_bool = in._bool;
-	_hex = in._hex;
-	if(type == V_STRING) _string = new string(*in._string);
+	_bool   = in._bool;
+	_hex    = in._hex;
+	if (type == V_STRING) {
+		_string = new string(*in._string);
+	}
```

Comment:
(I know this change is just reformatting) but looks like the `_string =` assignment will leak away any prior allocated `std::string` object it holds.

So we probably should check-and-free here, and should also assert that the incoming's `_string`'s pointer isn't `nullptr` (to avoid the dereference).

But before all that book-keeping to safely managed these damn pointers.. :sweat_smile: , I wonder how much trouble it would be to move `_string` to a  `std::string _string` instead of being a pointer?

Given `sizeof(std::string)` is only 8 bytes, it's not like the original pointer+`new` code is delaying the cost of constructing some massive wasteful class object that we might never use.

It would be really nice to get to:

``` C++
if (type == V_STRING) {
    _string = in._string;
}
```

Reply:
Okay, I've done all that, and some more. See the last commit.